apps/server/src/modules/learnroom/uc/course-info.uc.spec.ts

```diff
@@ -0,0 +1,215 @@
+				},
+			],
+		}).compile();
+
+		uc = module.get(CourseInfoUc);
+		authorizationService = module.get(AuthorizationService);
+		schoolService = module.get(SchoolService);
+		courseDoService = module.get(CourseDoService);
+		groupService = module.get(GroupService);
+		userService = module.get(UserService);
+		classService = module.get(ClassService);
+		groupRepo = module.get(GroupRepo);
+		classesRepo = module.get(ClassesRepo);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('getCourseInfo', () => {
+		const setup = () => {
+			const user = userFactory.withRoleByName(RoleName.TEACHER).buildWithId();
+			const teacher = userDoFactory.build({ id: user.id, firstName: 'firstName', lastName: 'lastName' });
+			const { adminUser } = UserAndAccountTestFactory.buildAdmin({}, [
+				Permission.COURSE_ADMINISTRATION,
+				Permission.ADMIN_VIEW,
+			]);
+			const group = groupFactory.build({ name: 'groupName' });
+			const clazz = classFactory.build({ name: 'A', gradeLevel: 1 });
+
+			const courses = courseDoFactory.buildList(5, {
+				syncedWithGroup: group.id,
+				teacherIds: [user.id],
+				groupIds: [group.id],
+				classIds: [clazz.id],
+			});
+			const pagination = { skip: 1, limit: 2 };
+			const courseStatusQueryType: CourseStatusQueryType = CourseStatusQueryType.CURRENT;
+			const sortByField: CourseSortQueryType = CourseSortQueryType.NAME;
+			const sortOrder: SortOrder = SortOrder.asc;
+
+			const school = schoolFactory.build();
+			schoolService.getSchoolById.mockResolvedValueOnce(school);
+			authorizationService.getUserWithPermissions.mockResolvedValue(adminUser);
+			authorizationService.checkPermission.mockReturnValueOnce(undefined);
+			groupRepo.findGroupById.mockResolvedValue(group);
+			groupService.findById.mockResolvedValue(group);
+			userService.findById.mockResolvedValue(teacher);
+			classService.findById.mockResolvedValue(clazz);
+			classesRepo.findClassById.mockResolvedValue(clazz);
+
+			return {
+				user,
+				courses,
+				pagination,
+				school,
+				adminUser,
+				group,
+				courseStatusQueryType,
+				sortByField,
+				sortOrder,
+				clazz,
+			};
+		};
+
+		it('should return courses with sorted and filtered results', async () => {
+			const {
+				courses,
+				clazz,
+				group,
+				school,
+				adminUser,
+				sortByField,
+				courseStatusQueryType: statusTypeQuery,
+				pagination,
+				sortOrder,
+				user,
+			} = setup();
+			courseDoService.getCourseInfo.mockResolvedValueOnce(new Page<Course>(courses, 5));
+
+			const result: Page<CourseInfoDto> = await uc.getCourseInfo(
+				adminUser.id,
+				school.id,
+				sortByField,
+				statusTypeQuery,
+				pagination,
+				sortOrder
+			);
+
+			const filter = { schoolId: school.id, courseStatusQueryType: statusTypeQuery };
+			const options = {
+				pagination,
+				order: {
+					[sortByField]: sortOrder,
+				},
+			};
+
+			expect(schoolService.getSchoolById).toHaveBeenCalledWith(school.id);
```

Comment:
technically each it should only have one expect, like "it should check the permission", "it should load the correct user" and "it should return the course info". Idk if someone already gave you the coding documentation, but here is a link to the testing guidelines https://documentation.dbildungscloud.dev/docs/schulcloud-server/Coding-Guidelines/testing